### PR TITLE
Fixed lower-triangular matrices

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: numerics/fixed_arrays.hpp [hunk 1]
```diff
@@ -65,6 +65,26 @@ FixedVector<Product<ScalarLeft, ScalarRight>, rows> operator*(
...
+  // The |data| must be in row-major format.
+  explicit FixedStrictlyLowerTriangularMatrix(
+      std::array<Scalar, kDimension> const& data);
+  FixedStrictlyLowerTriangularMatrix(
+      std::initializer_list<Scalar> const& data);  // NOLINT(runtime/explicit)
+
+  // For  0 < j < i < rows, the entry is a_ij accessed as |a[i][j]|.
+  // if i and j do not satisfy these conditions, the expression |a[i][j]| is
+  // erroneous.
+  Scalar const* operator[](int i) const;
```
Comment (pleroy) [inline numerics/fixed_arrays.hpp]: `const` all the ints.  Also `i` -> `index` for consistency with existing code.

File: numerics/fixed_arrays.hpp [hunk 2]
```diff
@@ -65,6 +65,26 @@ FixedVector<Product<ScalarLeft, ScalarRight>, rows> operator*(
...
+class FixedStrictlyLowerTriangularMatrix {
+ public:
+  static int const kDimension = rows * (rows - 1) / 2;
+  // The |data| must be in row-major format.
+  explicit FixedStrictlyLowerTriangularMatrix(
+      std::array<Scalar, kDimension> const& data);
+  FixedStrictlyLowerTriangularMatrix(
+      std::initializer_list<Scalar> const& data);  // NOLINT(runtime/explicit)
+
+  // For  0 < j < i < rows, the entry is a_ij accessed as |a[i][j]|.
```
Comment (pleroy) [inline numerics/fixed_arrays.hpp]: In the wrong order these words are.

File: numerics/fixed_arrays_body.hpp [hunk 3]
```diff
@@ -99,5 +99,30 @@ FixedVector<Product<ScalarLeft, ScalarRight>, rows> operator*(
...
+template<typename Scalar, int rows>
+FixedStrictlyLowerTriangularMatrix<Scalar, rows>::
+    FixedStrictlyLowerTriangularMatrix(
+        std::initializer_list<Scalar> const& data) {
+  CHECK_EQ(kDimension, data.size());
+  std::copy(data.begin(), data.end(), data_.begin());
+}
+
+template<typename Scalar, int rows>
+Scalar*  FixedStrictlyLowerTriangularMatrix<Scalar, rows>::operator[](int i) {
```
Comment (pleroy) [inline numerics/fixed_arrays_body.hpp]: `s/  / /`

File: numerics/fixed_arrays_test.cpp [hunk 4]
```diff
@@ -5,21 +5,26 @@
...
+           5, 8, 13}) {}
 
   FixedMatrix<double, 3, 4> m34_;
   FixedVector<double, 3> v3_;
   FixedVector<double, 4> v4_;
+  FixedStrictlyLowerTriangularMatrix<double, 4> l4_;
 };
 
-TEST_F(ArraysTest, Assignment) {
+TEST_F(FixedArraysTest, Assignment) {
```
Comment (pleroy) [inline numerics/fixed_arrays_test.cpp]: Add a test here for assigning/creating FSLTMs.

File: numerics/fixed_arrays_test.cpp [hunk 5]
```diff
@@ -5,21 +5,26 @@
...
-  ArraysTest()
+  FixedArraysTest()
     : m34_({-8, -6, -4, -7,
             -4, -10, 9, -5,
              6, -3, -2, -9}),
       v3_({10, 31, -47}),
-      v4_({-3, -3, 1, 4}) {}
+      v4_({-3, -3, 1, 4}),
+      l4_({
+           1,
```
Comment (pleroy) [inline numerics/fixed_arrays_test.cpp]: I'd put the 1 on the same line as the {.

File: numerics/fixed_arrays_body.hpp (hunk 3 above)
Comment (eggrobin) [inline numerics/fixed_arrays_body.hpp]: Done.

File: numerics/fixed_arrays.hpp (hunk 2 above)
Comment (eggrobin) [inline numerics/fixed_arrays.hpp]: Done.

File: numerics/fixed_arrays.hpp (hunk 1 above)
Comment (eggrobin) [inline numerics/fixed_arrays.hpp]: Done.

**Pull request merged**